### PR TITLE
docs: validate that the docs can be parsed by mdx

### PR DESCRIPTION
**Pull request by ethomson**
Although our documentation is rendered for the "in the box" docs by cmark-gfm, the upstream docs site (docs.npmjs.com) uses mdx, which is a much stricter parser.  Update our docs generator site to ensure that mdx can parse our documentation as well, to ensure that we get fast feedback when it would fail.

### CODE REVIEW

File: docs/dockhand.js [hunk 1]
```diff
@@ -225,3 +246,11 @@ function headerLevel(node) {
 function debug(str) {
     console.log(str);
 }
+
+class MarkdownError extends Error {
+    constructor(file, inner) {
+        super(`failed to parse ${file}`);
```
Comment (wraithgar) [inline docs/dockhand.js:252]: Do we care about the original error being eaten here?

Comment (ljharb) [reply docs/dockhand.js:252]: This is a great use case for the error cause proposal, but lacking that, you could make MarkdownError extend AggregateError instead, and preserve the original error that way?

Comment (darcyclarke) [reply docs/dockhand.js:252]: @ljharb @wraithgar I don't think we care that much about this until we actually run into a problem making the docs pass/build. We can backlog some work to improve this & feel free to add changes.

**Review by darcyclarke — APPROVED** [review]
LGTM

**Pull request merged**